### DISCUSSION TITLE
where main coding ?

### DISCUSSION THREAD

**Discussion by koi0823** [trackmania-rl/tmrl#107] [General]
can i ask a question about where the coding for main coding to run the program

![image](https://github.com/trackmania-rl/tmrl/assets/86875051/d6f4ecc6-c4d1-42f1-8668-f98039312fb4)
is it only this 

but i customer map so i quite confuse which is the main coding i wanna to debug it

**Comment by shaneaflyz** [comment]
same to me 
i didnt clone the repo but i can run tmrl in cmd 
why is that ?


@yannbouteiller

**Reply by yannbouteiller** [reply]
When you install tmrl with pip, it installs the library in your active python environment and you can then launch tmrl commands using python with the -m option, as in `python3 -m tmrl --test`.

These commands are defined in `__main__.py`

**Reply by shaneaflyz** [reply]
how do i run it locally with the code ?
@yannbouteiller

**Reply by yannbouteiller** [reply]
You can clone the repo and install it locally by executing `pip install -e .` where the `setup.py` file is.
